fw/cfg.c

```diff
@@ -1384,7 +1385,7 @@ _parse_integer(const char *s, unsigned int base, unsigned long *p)
 		unsigned int lc = c | 0x20; /* don't tolower() this line */
 		unsigned int val;
 
-		if ('0' <= c && c <= '9')
+		if (IS_DIGIT(c))
```

Comment:
If you move the comparison to a macro, then use the Hacker's Delight trick like `v - ('a' – 0x80) < 'z' - 'a' + 1 - 0x80` depending whether `c` is signed or not

Reply:
After reading book for a while, I can't quite understand the code above (`0x80` part).

Is this code OK?
```
#define IS_DIGIT(c) ((unsigned char) (c - '0') <= (unsigned char) ('9' - '0'))
```